nflow-engine/src/main/java/io/nflow/engine/service/WorkflowInstanceService.java

```diff
@@ -55,7 +55,20 @@ public WorkflowInstanceService(WorkflowInstanceDao workflowInstanceDao, Workflow
    * @throws EmptyResultDataAccessException If workflow instance is not found.
    */
   public WorkflowInstance getWorkflowInstance(long id, Set<WorkflowInstanceInclude> includes, Long maxActions) {
-    return workflowInstanceDao.getWorkflowInstance(id, includes, maxActions);
+    return getWorkflowInstance(id, includes, maxActions, false);
```

Comment:
How about just adding the new parameter here? We are breaking backwards compatibility anyway in 7.0.0

Reply:
7.0.0 was already released, bygones

Reply:
8.0.0 is not released yet, so this comment holds again - I would just add the parameter instead of adding the compatibility wrapper that hides the value of queryArchive used

Reply:
Yeah that is a valid point, we can keep it